PC/winreg.c

```diff
@@ -1463,6 +1476,8 @@ winreg_QueryInfoKey_impl(PyObject *module, HKEY key)
     return ret;
 }
 
+#ifndef MS_WINDOWS_GAMES
```

Comment:
Here's an implementation of `winreg.QueryValue()` that's based on `RegOpenKeyExW()` and `RegQueryValueExW()`.

```c
static PyObject *
winreg_QueryValue_impl(PyObject *module, HKEY key, const Py_UNICODE *sub_key)
/*[clinic end generated code: output=c655810ae50c63a9 input=41cafbbf423b21d6]*/
{
    LONG rc;
    HKEY childKey;
    WCHAR buf[256], *pbuf = buf;
    DWORD size = sizeof(buf);
    DWORD type;
    Py_ssize_t length;
    PyObject *result = NULL;

    if (PySys_Audit("winreg.QueryValue", "nuu",
                    (Py_ssize_t)key, sub_key, NULL) < 0) {
        return NULL;
    }

    if (key == HKEY_PERFORMANCE_DATA) {
        return PyErr_SetFromWindowsErrWithFunction(ERROR_INVALID_HANDLE,
                                                   "RegQueryValue");
    }

    if (!sub_key || !sub_key[0]) {
        childKey = key;
    }
    else {
        Py_BEGIN_ALLOW_THREADS
        rc = RegOpenKeyExW(key, sub_key, 0, KEY_QUERY_VALUE, &childKey);
        Py_END_ALLOW_THREADS
        if (rc != ERROR_SUCCESS) {
            return PyErr_SetFromWindowsErrWithFunction(rc, "RegOpenKeyEx");
        }
    }

    while (1) {
        Py_BEGIN_ALLOW_THREADS
        rc = RegQueryValueExW(childKey, NULL, NULL, &type, (LPBYTE)pbuf,
                              &size);
        Py_END_ALLOW_THREADS
        if (rc != ERROR_MORE_DATA) {
            break;
        }
        void *tmp = PyMem_Realloc(pbuf != buf ? pbuf : NULL, size);
        if (tmp == NULL) {
            PyErr_NoMemory();
            goto exit;
        }
        pbuf = tmp;
    }

    if (rc == ERROR_SUCCESS) {
        if (type != REG_SZ) {
            PyErr_SetFromWindowsErrWithFunction(ERROR_INVALID_DATA,
                                                "RegQueryValue");
            goto exit;
        }
        length = wcsnlen(pbuf, size / sizeof(WCHAR));
    }
    else if (rc == ERROR_FILE_NOT_FOUND) {
        // Return an empty string if there's no default value.
        length = 0;
    }
    else {
        PyErr_SetFromWindowsErrWithFunction(rc, "RegQueryValueEx");
        goto exit;
    }

    result = PyUnicode_FromWideChar(pbuf, length);

exit:
    if (pbuf != buf) {
        PyMem_Free(pbuf);
    }
    if (childKey != key) {
        RegCloseKey(childKey);
    }
    return result;
}
```